File: api/api.web.model/BoardWebModel.cs
namespace Apex.Api.Web.Model
{
    public class LocationDto
    {
        public int Region { get; set; }
        public int X { get; set; }
        public int Y { get; set; }
    }

    public class CellDto
    {
        public int Id { get; set; }
        public LocationDto[] Locations { get; set; }
    }

    public class BoardDto
    {
        public int RegionCount { get; set; }
        public int RegionSize { get; set; }
        public CellDto[] Cells { get; set; }
    }
}

File: api/api.web/Model/BoardWebModel.fs
namespace Apex.Api.Web.Model

open System.ComponentModel.DataAnnotations

type LocationDto = {
    [<Required>]
    region : int
    
    [<Required>]
    x : int
    
    [<Required>]
    y : int
}

type CellDto = {
    [<Required>]
    id : int

    [<Required>]
    locations : List<LocationDto>
}

type BoardDto = {
    [<Required>]
    regionCount : int

    [<Required>]
    regionSize : int
    
    [<Required>]
    cells : List<CellDto>
}